db/seeds.rb

```diff
@@ -1,70 +1,78 @@
-    { first_name: 'User3', last_name: Faker::Name.last_name, email: "user3@aol.com", password: "password", role: "member" },
-    { first_name: 'Applicant1', last_name: Faker::Name.last_name, email: "applicant1@aol.com", password: "password", role: "applicant" },
-    { first_name: 'Applicant2', last_name: Faker::Name.last_name, email: "applicant2@aol.com", password: "password", role: "applicant" },
-    { first_name: 'Applicant3', last_name: Faker::Name.last_name, email: "applicant3@aol.com", password: "password", role: "applicant" },
-  ]
+  user_data = []
+  # create admin and mod data
+  2.times do |idx|
+    user_data.push(generate_user_data(role: 'admin', idx:))
+    user_data.push(generate_user_data(role: 'moderator', idx:))
+  end
+
+  # create member data
+  4.times do |idx|
+    user_data.push(generate_user_data(role: 'member', idx:))
+  end
 
-  users = []
+  # create applicant data
+  24.times do |idx|
+    user_data.push(generate_user_data(role: 'applicant', idx:))
+  end
 
   User.transaction do
     user_data.each do |data|
-      user = User.create!(data)
-      users << user
-      puts "User #{user.email} created"
+      User.create!(data)
+      puts "User #{data[:email]} created"
     end
   end
-  puts "Users seeded successfully!"
-
-  puts "Seeding pair requests..."
-  pair_request_data = [
-    { status: "pending", author: users[1], invitee: users[2], when: 1.day.from_now, duration: 30.minutes },
-    { status: "pending", author: users[1], invitee: users[3], when: 1.day.from_now, duration: 30.minutes },
-    { status: "pending", author: users[2], invitee: users[3], when: 1.day.from_now, duration: 45.minutes },
-    { status: "rejected", author: users[1], invitee: users[2], when: 1.day.from_now, duration: 30.minutes },
-    { status: "accepted", author: users[1], invitee: users[2], when: 5.day.from_now, duration: 60.minutes },
-    { status: "accepted", author: users[2], invitee: users[3], when: Time.now, duration: 60.minutes },
-    { status: "accepted", author: users[2], invitee: users[3], when: Time.now, duration: 60.minutes },
-    { status: "accepted", author: users[2], invitee: users[3], when: Time.now, duration: 60.minutes },
-    { status: "accepted", author: users[3], invitee: users[2], when: Time.now, duration: 60.minutes },
-    { status: "accepted", author: users[3], invitee: users[2], when: Time.now, duration: 60.minutes },
-    { status: "rejected", author: users[3], invitee: users[1], when: 20.days.from_now, duration: 15.minutes },
-  ]
+  users = User.all
+  puts 'Users seeded successfully!'
+
+  puts 'Seeding pair requests...'
+  pair_request_data = []
+  5.times do
+    pair_request_data.push(generate_pair_request_data(status: 'pending', users:))
+    pair_request_data.push(generate_pair_request_data(status: 'rejected', users:))
+    pair_request_data.push(generate_pair_request_data(status: 'accepted', users:))
```

Comment:
Awesome work in this file :raised_hands:  It was growing unwieldy to work in.

One thing I did want to draw attention to is that if you look at the old `accepted` pair requests, they all have a `when` time of *now*. This is for a specific reason. If you check the [policy for marking requests as complete](https://github.com/agency-of-learning/PairApp/blob/main/app/policies/pair_request/completion_policy.rb#L5), they prevent the user from doing it until the pair request has started.

Your current strategy starts all generated pair requests at some time in the future (a random sample between 1 to 4 days from now), which imo, introduces some friction for moving through the pair request lifecycle locally.

Reply:
I'll fix that up.